prometheus_client/parser.py

```diff
@@ -104,34 +215,50 @@ def _parse_value_and_timestamp(s: str) -> Tuple[float, Optional[float]]:
     values = [value.strip() for value in s.split(separator) if value.strip()]
     if not values:
         return float(s), None
-    value = float(values[0])
-    timestamp = (float(values[-1]) / 1000) if len(values) > 1 else None
+    value = parse_value(values[0])
+    timestamp = (parse_value(values[-1]) / 1000) if len(values) > 1 else None
     return value, timestamp
 
 
-def _parse_sample(text: str) -> Sample:
-    # Detect the labels in the text
+def parse_value(value):
```

Comment:
```suggestion
def _parse_value(value):
```
here as well to match `_parse_sample`